### PR TITLE
[utils/build-script] Introduce '--extra-cmake-vars' option

### PR DESCRIPTION
**Pull request by akyrtzi**
Introduce '--extra-cmake-vars' to allow setting any CMake veriables when configuring the targets.

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci  please test

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by akyrtzi** [comment]
@gribozavr please review

File: utils/build-script [hunk 1]
```diff
@@ -701,6 +701,13 @@ the number of parallel build jobs to use""",
         action="append", dest="extra_swift_args", default=[])
 
     parser.add_argument(
+        "--extra-cmake-vars", help=textwrap.dedent("""
```
Comment (gribozavr) [inline utils/build-script]: We call them 'options' throughout the build-script, so it would me more consistent to call it `--extra-cmake-options`

**Comment by gribozavr** [comment]
LGTM modulo a minor comment.

**Comment by gottesmm** [comment]
Don't we already have something like this in build-script-impl? Maybe see if you can rip that out as well? (Just search for cmake).

**Comment by gribozavr** [comment]
I looked for it, but only could find one for lldb.

**Comment by akyrtzi** [comment]
> We call them 'options' throughout the build-script, so it would me more consistent to call it --extra-cmake-options
There is a subtle difference, the cmake 'options' throughout the build-script-impl are of the full form that includes the dash and 'D', like "-DLLDB_ALLOW_STATIC_BINDINGS=1". With this option you are setting variable names like this:
--extra-cmake-vars=LLDB_ALLOW_STATIC_BINDINGS=1
Note lack of '-D'.

**Comment by gribozavr** [comment]
I see.  Do we really need to have this subtle difference?

**Comment by akyrtzi** [comment]
Probably not, I can change it to
--extra-cmake-options=-DLLDB_ALLOW_STATIC_BINDINGS=1
if you prefer.

**Comment by gribozavr** [comment]
I'd prefer that, it seems to be less magical.

**Comment by akyrtzi** [comment]
@swift-ci Please test

**Comment by rintaro** [comment]
@akyrtzi @gribozavr 
We already have [`--user-config-args`](https://github.com/apple/swift/blob/abecb2f6a92c27777d9b96322bb17ce4fd1348b7/utils/build-script-impl#L90) that is unconditionally added to the [*tail* of `cmake_options`](https://github.com/apple/swift/blob/abecb2f6a92c27777d9b96322bb17ce4fd1348b7/utils/build-script-impl#L2078), whereas `COMMON_CMAKE_OPTIONS` are normally prepended to product specific options.

Since, IIRC, when duplicated `-D` are specified, `cmake` uses latter values,  I think, `user-config-args` are more usable for this purpose.

**Comment by akyrtzi** [comment]
Ah I missed this, thanks for pointing it out!
How about we rename that to '--extra-cmake-options', '--user-config-args' seems rather vague to me.

**Comment by gribozavr** [comment]
That seems reasonable.  Could you keep the old option parseable, and produce a loud error though?  We'd remove it some time later.

**Pull request merged**